cmd/bootstrap/cmd/rootblock_test.go

```diff
@@ -131,3 +131,56 @@ func TestInvalidRootBlockViewSubprocess(t *testing.T) {
 		rootBlock(nil, nil)
 	})
 }
+
+// TestInvalidKVStoreValues verifies that running
+// rootBlock with an invalid kvstore values (default) on "main" or "testnet" chains.
+// The test runs in subprocesses because the tested code calls os.Exit.
+func TestInvalidKVStoreValues(t *testing.T) {
+	for _, chain := range []string{"main", "test"} {
+		t.Run("invalid kv store values for "+chain, func(t *testing.T) {
+			expectedError := fmt.Sprintf("cannot use default KVStore values (epoch extension view count and finalization safety threshold) on %q chain", chain)
+			extraEnv := []string{"CHAIN=" + chain}
+
+			runTestInSubprocessWithError(
+				t,
+				"TestInvalidKVStoreValuesSubprocess",
+				expectedError,
+				extraEnv,
+			)
+		})
+	}
+}
+
+// TestInvalidKVStoreValuesSubprocess runs in subprocess for invalid kvstore values test on various chains
+func TestInvalidKVStoreValuesSubprocess(t *testing.T) {
+	if os.Getenv("FLAG_RUN_IN_SUBPROCESS_ONLY") != "1" {
+		return
+	}
+
+	utils.RunWithSporkBootstrapDir(t, func(bootDir, partnerDir, partnerWeights, internalPrivDir, configPath string) {
+		setupHappyPathFlags(bootDir, partnerDir, partnerWeights, internalPrivDir, configPath)
+		flagFinalizationSafetyThreshold = defaultFinalizationSafetyThreshold
+		flagEpochExtensionViewCount = defaultEpochExtensionViewCount
+		flagRootChain = os.Getenv("CHAIN")
+
+		hook := zeroLoggerHook{logs: &strings.Builder{}}
+		log = log.Hook(hook)
+
+		rootBlock(nil, nil)
+	})
+}
+
+// runTestInSubprocessWithError executes a test function in a subprocess,
+// expecting it to fail with an error. It is used for testing code paths
+// that call os.Exit.
+func runTestInSubprocessWithError(t *testing.T, testName, expectedOutput string, extraEnv []string) {
```

Comment:
I believe we already have a similar function in previous [PR](https://github.com/onflow/flow-go/pull/7666/files#diff-110f7738ec280b0c313d32d3b6a38b5cddbf05ad71aac8adfa997a99195459deR118). Is it possible to combine them?